DM-43103: Add ObservationInfo.can_see_sky property

timj opened the pull request.

timj committed:
Modernize ruff configuration

timj committed:
Add base can_see_sky property

timj committed:
Fixes founds by ruff

timj committed:
Update python version in black usage

timj committed:
Allow translator subclass to specify list of on-sky observation types

timj committed:
Add can_see_sky observation types for HSC

timj committed:
Update setup-python and checkout action versions

timj committed:
Use newer codecov action

timj committed:
Allow default output stream to be None

This lets click override stdout for testing, which it can't
do if sys.stdout is forced to be the default.

timj committed:
Remove internal errstream parameter and replace with logging

Defaulting to sys.stderr hides it from click. Rather than
trying to use click.echo replace it with a logger (which
will default to using sys.stderr).

timj committed:
Fix logger usage in main cli

timj committed:
Add tests for the click command line

timj committed:
Add logging and trap no files found in write-sidecar CLI

erykoff reviewed (approved):
Looks fine, but I am wondering about the few error cases that coverage says aren't covered.  Also, does this have any effect on the race condition that we see sometimes in the tests?  (That may have been fixed elsewhere, I don't recall...)

timj commented:
I added 5% coverage so I'm going to call that a win.

This has no effect on the test failure we were getting. On a previous PR I made the test more expressive in its failure output so we would know why it is failing but I don't think it's failed since I added that.

timj committed:
Add some coverage configuration

This will stop it complaining about some lines that are never
going to be hit in a test.

timj commented:
Ok. I added extra coverage and stopped it complaining about some lines that were never going to be tested.
